front/lib/front_web/controllers/project_controller.ex

```diff
@@ -677,21 +678,38 @@ defmodule FrontWeb.ProjectController do
     branch = params["branch"] || project.repo_default_branch
     commit_sha = params["commit_sha"]
 
-    workflow =
-      Models.Workflow.find_latest(
-        project_id: project.id,
-        branch_name: branch,
-        commit_sha: commit_sha
-      )
-
     workflow_url =
-      if workflow do
-        workflow_path(conn, :show, workflow.id, pipeline_id: workflow.root_pipeline_id)
-      else
-        nil
+      if params["commit_sha"] != "" do
+        workflow =
+          Models.Workflow.find_latest(
+            project_id: project.id,
+            branch_name: branch,
+            commit_sha: commit_sha
+          )
+
+        if workflow do
+          workflow_path(conn, :show, workflow.id, pipeline_id: workflow.root_pipeline_id)
+        end
       end
 
-    conn |> put_status(200) |> json(%{workflow_url: workflow_url})
+    artifact_url =
+      if params["job_id"] do
+        job = Job.find(params["job_id"])
+
+        if job && job.state == "passed" do
+          {:ok, url} =
+            Artifacthub.signed_url(project.id, "jobs", job.id, ".workflow_editor/commit_sha.val")
+
+          url
+        end
+      end
+
+    conn
+    |> put_status(200)
+    |> json(%{
+      workflow_url: workflow_url,
+      artifact_url: artifact_url
+    })
```

Comment:
There is already this [check_commit_job](https://github.com/semaphoreio/semaphore/blob/main/front/lib/front_web/controllers/project_controller.ex#L538) endpoint for this check, why not use that one?

Reply:
The only reason is that we're already using this endpoint in the project onboarding - it made the integration with the UI a little bit easier.